tests/ClockMockTest.php

```diff
@@ -61,7 +61,16 @@ public function test_DateTimeImmutable_createFromFormat()
         // Verification: when not provided with a time, createFromFormat should use current time.
         $this->assertSame('2022-05-28 12:13:14', $dateTimeFromFormat->format('Y-m-d H:i:s'));
     }
-    
+
+    public function test_DateTime_createFromFormat_epoch()
+    {
+        ClockMock::freeze(new \DateTimeImmutable('1986-06-05 12:13:14'));
+
+        $dateTimeFromFormat = \DateTime::createFromFormat('U', strval(time()));
+
+        $this->assertSame('1986-06-05 12:13:14', $dateTimeFromFormat->format('Y-m-d H:i:s'));
+    }
```

Comment:
This test is passing even without your fix. I believe this should not be included as it's basically a duplicate of `test_DateTime_createFromFormat`.

What you want to test here is that, when full time (including microseconds) is provided in `createFromFormat`, it works as expected (because right now, without the fix, it doesn't).

My suggestion would be to rename the existing `test_DateTime_createFromFormat` into `test_DateTime_createFromFormat_without_time` and add another one `test_DateTime_createFromFormat_with_time`, which is basically the same as the former, but with the DateTime object created with a format like `'Y-m-d H:i:s.u'`. What do you think?